service/src/main/java/bio/terra/catalog/service/DatasetService.java

```diff
@@ -135,21 +136,32 @@ private List<DatasetResponse> collectDatasets(StorageSystem system) {
     return createDatasetResponses(roleMap, system);
   }
 
+  static class ContextCopier {
```

Comment:
I would call this something around `ParallelRequestWithHttpContextCopying` or something? My only concern is remembering the _purpose_ of the class when we come back to it. (I know there's the comment on the method, but I think on the class would be good.

Reply:
I'll put it out to its own class and will try a different name there.

I'd like to find a shorter name than `ParallelRequestWithHttpContextCopying`. I think `http` is not needed since the request context can contain other things besides http state. And the "parallel with" is part of the operation itself I think. Maybe `RequestContextCopier.parallelWithContext()`? Although that says `Context` twice which seems redundant too...

Reply:
If you really want to avoid `Parallel` being in there, I think that this is probably the best name. But yeah, the question in my mind is whether we will understand what this class is for in the future. Because it is just used in this class, it could potentially be a static method in this class, rather than a new class. That may be clearer in this case, in my opinion, just because my confusion is mostly around the intermediate naming for the class.